### PR TITLE
[BugFix] Fix checking the existance of domained users fails

### LINKED ISSUE #10997: grant/revoke role to a domain user will fail
```
GRANT role1 TO zhangsan@['starrocks.domain1']
```

will cause error.

```
user zhangsan@['starrocks.domain1'] not exist!
```

Though we are not about to support  domain user in the future, we still have to fix this in consideration of our current users.

### StarRocks version (Required)
2.3+

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #10997 

## Problem Summary(Required) ：

My original code checks if a user exits by looking through all the `GlobalPrivEntry` in `UserPrivTable`. But domained user won't store in `UserPrivTable`, it should be checked through `PropertyManager`. Luckily, there is already a function called `Auth.doesUserExist` that has done this job. I'm replacing `Auth.userPrivTable.doesUserExist` with `Auth.doesUserExist` to fix this bug.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|1|1|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — DISMISSED** [review]

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/mysql/privilege/Auth.java [hunk 1]
```diff
@@ -1097,7 +1097,7 @@ private void grantImpersonateToRoleInternal(
     }
 
     // return true if user ident exist
-    private boolean doesUserExist(UserIdentity userIdent) {
+    public boolean doesUserExist(UserIdentity userIdent) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/mysql/privilege/Auth.java:1100]: It's better to add a more concise message about the bug. The problem and solution should be described clearly.

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/mysql/privilege/Auth.java:1100]: done

**Review by chaoyli — APPROVED** [review]

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.4 branch-2.3
¡Muchas gracias!

**Pull request merged**